Flepper.QueryBuilder/Operators/Comparison/ComparisonOperators.cs

```diff
@@ -42,6 +47,11 @@ public IComparisonOperators LessThanOrEqualTo(int value)
 
         public IComparisonOperators NotEqualTo(object value)
         {
+            if (value == null)
+            {
+                Command.AppendFormat("IS NOT NULL ");
+                return this;
```

Comment:
Instead of returning this, do an if - else

```csharp
if (value is null) 
    Command.AppendFormat("IS NOT NULL "); 
else
    Command.AppendFormat("= {0} ", value.InsertQuotationMarksIfDateOrString()); 
```

Reply:
@AlbertoMonteiro I'm not do that because in this way we have more cyclomatic complexity on code.

Follow the changes and the cyclomatic complexity to both cases. See before it is 2 and with new method is 3:
! [image] (https://user-images.githubusercontent.com/889634/30355026-f5174fbc-9806-11e7-96ba-56a9765c2ea4.png)

Reply:
@reinaldocoelho its how your analyser "understeand" that, but both code have 2 only paths.

Lets analyse the Intermediate language from both methods:

The EqualTo Method that use 2 returns:

```
.method public final hidebysig newslot virtual 
	instance class Flepper.QueryBuilder.IComparisonOperators EqualTo (
		object 'value'
	) cil managed 
{
	// Header Size: 1 byte
	// Code Size: 47 (0x2F) bytes
	.maxstack 8
	IL_0000: ldarg.1
	IL_0001: brtrue.s  IL_0016
	IL_0003: ldarg.0
	IL_0004: ldfld     class [netstandard]System.Text.StringBuilder Flepper.QueryBuilder.Base.BaseQueryBuilder::Command
	IL_0009: ldstr     "IS NULL "
	IL_000E: callvirt  instance class [netstandard]System.Text.StringBuilder [netstandard]System.Text.StringBuilder::Append(string)
	IL_0013: pop
	IL_0014: ldarg.0
	IL_0015: ret
	IL_0016: ldarg.0
	IL_0017: ldfld     class [netstandard]System.Text.StringBuilder Flepper.QueryBuilder.Base.BaseQueryBuilder::Command
	IL_001C: ldstr     "= {0} "
	IL_0021: ldarg.1
	IL_0022: call      object Flepper.QueryBuilder.Utils.Extensions.ValuesExtensions::InsertQuotationMarksIfDateOrString(object)
	IL_0027: callvirt  instance class [netstandard]System.Text.StringBuilder [netstandard]System.Text.StringBuilder::AppendFormat(string, object)
	IL_002C: pop
	IL_002D: ldarg.0
	IL_002E: ret
} // end of method ComparisonOperators::EqualTo
```

The NotEqualTo that use IF/ELSE

```
.method public final hidebysig newslot virtual 
	instance class Flepper.QueryBuilder.IComparisonOperators NotEqualTo (
		object 'value'
	) cil managed 
{
	// Header Size: 1 byte
	// Code Size: 47 (0x2F) bytes
	.maxstack 8
	IL_0000: ldarg.1
	IL_0001: brtrue.s  IL_0016
	IL_0003: ldarg.0
	IL_0004: ldfld     class [netstandard]System.Text.StringBuilder Flepper.QueryBuilder.Base.BaseQueryBuilder::Command
	IL_0009: ldstr     "IS NOT NULL "
	IL_000E: callvirt  instance class [netstandard]System.Text.StringBuilder [netstandard]System.Text.StringBuilder::Append(string)
	IL_0013: pop
	IL_0014: br.s      IL_002D
	IL_0016: ldarg.0
	IL_0017: ldfld     class [netstandard]System.Text.StringBuilder Flepper.QueryBuilder.Base.BaseQueryBuilder::Command
	IL_001C: ldstr     "<> {0} "
	IL_0021: ldarg.1
	IL_0022: call      object Flepper.QueryBuilder.Utils.Extensions.ValuesExtensions::InsertQuotationMarksIfDateOrString(object)
	IL_0027: callvirt  instance class [netstandard]System.Text.StringBuilder [netstandard]System.Text.StringBuilder::AppendFormat(string, object)
	IL_002C: pop
	IL_002D: ldarg.0
	IL_002E: ret
} // end of method ComparisonOperators::NotEqualTo
```

You can see that those methods will differ on instruction: "IL_0014", the if/else will jump to last two lines, and the "2 returns" will repeat the 2 last lines.

Reply:
Let me show you another tool that calculates te complexity

![EqualTo](https://user-images.githubusercontent.com/836496/30356416-4931c642-980f-11e7-9545-6aaf97dfc7db.png)

![NotEquals](https://user-images.githubusercontent.com/836496/30356423-56197684-980f-11e7-9804-30575ae06be6.png)